src/main/java/com/codeborne/selenide/ConditionWaiter.java

```diff
@@ -0,0 +1,42 @@
+package com.codeborne.selenide;
+
+import org.awaitility.core.ConditionTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.function.Predicate;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+
+
+public class ConditionWaiter {
+
+  private static final Logger logger = LoggerFactory.getLogger(ConditionWaiter.class);
+
+  public static void waitFor(Conditional conditional, Predicate predicate, String message) {
+    waitFor(conditional, predicate, message, 10);
+  }
+
+  public static void waitFor(Conditional conditional, Predicate predicate, String message, int timeout) {
+    try {
+      await()
+        .pollInterval(1, SECONDS)
+        .timeout(timeout, SECONDS)
+        .pollInSameThread()
+        .until(() -> checkUnThrowable(conditional, predicate));
+    } catch (ConditionTimeoutException e) {
+      throw new ConditionTimeoutException(message);
```

Comment:
Re-wrap this exception with selenide one.

Reply:
The most appropriate that i found was SoftAssertionError, i used it. 
But i think we can create new on like ConditionNotMatchException)

Reply:
yes we have to create new exception like it is for element conditions.

Reply:
aded